src/function/table/arrow.cpp

```diff
@@ -422,9 +434,36 @@ static void SetVectorString(Vector &vector, idx_t size, char *cdata, T *offsets)
 	}
 }
 
+void DirectConversion(Vector &vector, ArrowArray &array, ArrowScanState &scan_state, idx_t nested_offset) {
+	auto data_ptr = (data_ptr_t)array.buffers[1] +
+	                GetTypeIdSize(vector.GetType().InternalType()) * (scan_state.chunk_offset + array.offset);
+	if (nested_offset != 0) {
+		data_ptr = (data_ptr_t)array.buffers[1] +
+		           GetTypeIdSize(vector.GetType().InternalType()) * (array.offset + nested_offset);
+	}
+	FlatVector::SetData(vector, data_ptr);
+}
+
+template <class T>
+void TimeConversion(Vector &vector, ArrowArray &array, ArrowScanState &scan_state, idx_t nested_offset, idx_t size,
```

Comment:
Could we use integer arithmetic here instead of `double conversion` to avoid potential problems with floating point inaccuracies? E.g. you could add another class to the templates:

```cpp
struct SecondsToMicros {
    static int64_t Convert(int32_t input) {
        return input * 1000000;
    }
};

struct MicroSecondsToMicros {
    static int64_t Convert(int32_t input) {
        return input * 1000;
    }
};
```

Then pass that in as a template parameter.